### PR TITLE
dkg: verify private key before p2p setup

### PR DESCRIPTION
**Pull request by corverroos**
This was only added to `charon run`, also doing this in dkg now.

category: feature
ticket: #937

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -675,31 +679,3 @@ func (h httpServeHook) Call(context.Context) error {
 
 	return nil
 }
-
-// verifyP2PKey returns an error if the p2pkey doesn't match any lock operator ENR.
-func verifyP2PKey(lock cluster.Lock, key *ecdsa.PrivateKey) error {
```
Comment (corverroos) [inline app/app.go:680]: moved this to p2p package so DKG can access it

File: p2p/peer.go [hunk 2]
```diff
@@ -139,3 +140,26 @@ func (p *MutablePeer) Subscribe(sub func(Peer)) {
...
+		if err != nil {
+			return errors.Wrap(err, "key to bytes")
+		}
+
+		if bytes.Equal(wantBytes, gotBytes) {
+			return nil
+		}
+	}
+
+	return errors.New("private key not matching any operator ENR") // This message seems to know something about the context in which it is used :(
```
Comment (xenowits) [inline p2p/peer.go:164]: i didn't understand the comment ;)

**Review by xenowits — APPROVED** [review]

File: p2p/peer.go (hunk 2 above)
Comment (corverroos) [reply p2p/peer.go]: How about: `unknown private key provided, it doesn't match any public key encoded inside the operator ENRs`

Comment (corverroos) [reply p2p/peer.go]: This functions isn't provided with any operators or ENRs so the fact that it mentions it in the error message is assuming specific implicit external use-case

**Pull request merged**